tests/test_vissharedmemreaderwriter.py

```diff
@@ -0,0 +1,242 @@
+# === Start Python 2/3 compatibility
+from __future__ import absolute_import, division, print_function, unicode_literals
+from future.builtins import *  # noqa pylint: disable=W0401, W0614
+from future.builtins.disabled import *  # noqa pylint: disable=W0401, W0614
+
+# == End Python 2/3 compatibility
+
+import logging
+import numpy as np
+import os
+import posix_ipc
+import pytest
+import re
+import signal
+from subprocess import Popen
+import shutil
+import tempfile
+import threading
+from time import sleep
+
+from comet import Manager
+
+from kotekan import runner, shared_memory_buffer
+
+# use tempfile creation to get exclusive random strings
+useless_file = tempfile.NamedTemporaryFile()
+fname_buf = "calBuffer_" + os.path.split(useless_file.name)[-1]
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+@pytest.fixture()
+def comet_broker():
+    broker_path = shutil.which("comet")
+    if not broker_path:
+        pytest.skip(
+            "Make sure PYTHONPATH is set to where the comet dataset broker is installed."
+        )
+
```

Comment:
This should test to see if `redis` is available too. See `has_redis` in `test_dataset_broker.py`